### ISSUE TITLE
[REQUEST] 8x16 sprite mode?

### ISSUE THREAD

**Issue by TheRealHamtaro126** [xverizex/RetroSpriteEditor#1]
Appearently the NES has a 8x16 sprite mode, of which this app did not use or did not take into the account as of now, so I suggest a implementation to add it, it is very good to have, if done soon.

Plus, for those using MMC5, I'd just make an additional "MMC5 8x16 Sprite Mode" in your options too... in addition to the normal NES mode usage.

This way, your app could be a REAL step up from the other alternatives used, such as NEXXT/NES Screen Tool, YY-CHR Screen Editor and Nostalgic Assets Workshop (NAW was free, Now costs an arm and a leg for no good reason...)

**Comment by xverizex** [comment]
Ok, I'll add 8x16 mode. In the first time, I wanted to add at least something to make it work.

**Comment by xverizex** [comment]
I haven't figured out how the 8x16 mode works in the NES yet.

**Comment by TheRealHamtaro126** [comment]
Have you looked in the NESDEV wiki?!?
https://www.nesdev.org/wiki/PPU_OAM

see PPU OAM -> Byte 1 for more information

If you want to implement 8x16 mode for MMC5, see thier wiki on the MMC5 mapper:
https://www.nesdev.org/wiki/MMC5

For the MMC5 side, you need to look at specific places in the mapper, specific to sprite CHR, not much.

**Comment by TheRealHamtaro126** [comment]
MMC5 is optional BTW...

**Comment by xverizex** [comment]
If I make the selection between 8x8 or 8x16 mode, it is well, or need to select MMC5 exactly? Or you want special parameters for MMC5?

**Comment by xverizex** [comment]
Can I question? If I select 8x16 mode, then the background is 8x16 mode too?

**Comment by xverizex** [comment]
I still haven't worked with bank switching.

**Comment by TheRealHamtaro126** [comment]
1: would like just a special 3-bank mode for MMC5, one using 2 banks for 8x16 PRG for each 4k bank, up to 8k sprites with 4k tileset for BG/FG tiles using just NES nametable mode, can be 12k CHR, or 16k for NES compatibility, with 4k CHR unused, altogether...

2: Can have either 8x16 and 8x8 BG modes, selection mode per user project type.

3: Do not need to have bankswitching, sortof, there is both 4k and 8k CHR modes for 8x16, for sprites, 4k is usable more than 8k CHR mode, but 8k can be still used in most cases for 8x16 in MMC5 if minimally needed...

in short for the 4k mode: but i dunno about 8k, so 4k is perferred if using minimal graphics!

**Comment by TheRealHamtaro126** [comment]
I can always write my own code into certain places using this app if 8x16 is implemented, even with MMC5, I can pretty much make some stuff for mappers myself pretty easily.

6502 Code may look hard, but isn't once you learn a bit of reading, hacking around, and skill...

**Comment by xverizex** [comment]
Can you give me your code  with MMC5 to be tested it?

**Comment by TheRealHamtaro126** [comment]
I referred to the fact that I can always make my own code for myself, while doing work on my own stuff, not need to know anything else other than the fact, other than this, you are gonna go through stuff some say... are obstacles in coding especially if new to 6502.

There is this: Patience is a virtue, there is always another way around things...

Think outside the box, Coding requires critical thoughts and decisions,

For me and you, Hope is all around us, Dont forget to take breaks... Get plenty of rest if needed.

**Comment by xverizex** [comment]
Ok. :)

**Comment by xverizex** [comment]
> 2: Can have either 8x16 and 8x8 BG modes, selection mode per user project type.

I don't correct understood you. )) For background I haven't found the mode 8x16, so there is mode 8x8 is available, right?

**Comment by xverizex** [comment]
The commit 1c9ac86 solve this issue...

I still haven't tested in nes game, but I think the implementation is correct. mmc still not implemented, and it will not be implemented yet.
